clang/lib/CIR/CodeGen/CIRGenExprCst.cpp

```diff
@@ -1238,6 +1245,28 @@ buildArrayConstant(CIRGenModule &CGM, mlir::Type DesiredType,
   return {};
 }
 
+mlir::TypedAttr ConstantEmitter::tryEmitPrivate(const Expr *E, QualType T) {
```

Comment:
Is the implementation of both these methods actually being used or were they just a workaround for building clangir?

Reply:
I used the implementation from clang's Codegen for the added methods

Reply:
That's fine, but we usually try not to add code unless we have tests for it, so the solution here would be to either (a) delete the other code in the body and add an assert or (b) leave the code there and still add an assert in the beginning. I'd prefer (a) unless you're already looking into a testcase that exercises (b).

Reply:
added asserts instead of implementation